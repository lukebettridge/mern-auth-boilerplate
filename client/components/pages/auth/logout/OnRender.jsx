import { useEffect } from "react";
import PropTypes from "prop-types";

const OnRender = props => {
	useEffect(() => {
		props.mutation();
	}, []);
	return null;
};

OnRender.propTypes = {
	mutation: PropTypes.func.isRequired
};

export default OnRender;
