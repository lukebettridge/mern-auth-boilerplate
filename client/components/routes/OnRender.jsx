import { useEffect } from "react";
import PropTypes from "prop-types";

const OnRender = props => {
	useEffect(() => {
		if (props.method) props.method();
	}, []);
	return null;
};

OnRender.propTypes = {
	method: PropTypes.func
};

export default OnRender;
