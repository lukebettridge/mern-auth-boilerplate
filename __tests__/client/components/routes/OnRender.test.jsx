import React from "react";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";

import OnRender from "components/routes/OnRender";

const method = jest.fn();

describe("OnRender component", () => {
	beforeEach(() => {
		jest.resetAllMocks();
	});

	it("snapshot renders", () => {
		const subject = mount(<OnRender method={method} />);
		expect(toJSON(subject)).toMatchSnapshot();
	});

	it("calls method prop on mount", () => {
		const subject = mount(<OnRender method={method} />);
		expect(method).toHaveBeenCalled();
		expect(subject).toEqual({});
	});
});
