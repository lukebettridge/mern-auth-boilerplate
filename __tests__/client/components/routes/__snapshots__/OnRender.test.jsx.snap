// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnRender component snapshot renders 1`] = `
<OnRender
  method={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;
